### DISCUSSION TITLE
XRDP connection to RHEL 8.5 single application (kiosk) mode

### DISCUSSION THREAD

**Discussion by rathbunr** [neutrinolabs/xrdp#2254] [Q&A] [answered]
I am trying to get XRDP working with Red Hats single application mode which is described here. https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/using_the_desktop_environment_in_rhel_8/assembly_restricting-the-session-to-a-single-application_using-the-desktop-environment-in-rhel-8

When I connect directly to the system it comes up in kiosk mode it works perfectly fine. As soon as I RDP into the RHEL system it puts me in a full GUI user session, which is not the behavior I am looking for. I read a similar article here based on Ubuntu, but am unsure how to resolve this issue on the RHEL platform. My goal is to use Red Hat instructions as much as possible and restrict the user to a single application.

**Comment by matt335672** [comment, marked as answer]
Interesting question.

I've installed `gnome-session-kiosk-session` on a CentOS 8 stream machine as instructed by your link (which uses `yum` rather than `dnf` at the time of writing!) and had a look at it.

Not a lot gets installed. The most significant bit appears to be the desktop file for the kiosk session:-

```ini
$ cat /usr/share/xsessions/com.redhat.Kiosk.desktop
[Desktop Entry]
Name=Kiosk
Comment=Kiosk mode
Exec=/usr/bin/gnome-session --session=redhat-kiosk
DesktopNames=Red-Hat-Kiosk;GNOME;
```

At a first glance it appears the thing to do is to start the session using the `Exec` line indicated above.

In your kiosk user, create a file `$HOME/.xsession` with the following contents:-

```bash
#!/bin/sh

exec /usr/bin/gnome-session --session=redhat-kiosk
```

Make it executable, and try logging in again.

**Comment by rathbunr** [comment]
I cannot create executable content in that location, I will try a symbolic link and let you know if that works out.

**Comment by matt335672** [comment]
I'm sure we can come up with other options, but this one seems the simplest to me.

**Comment by rathbunr** [comment]
Matt,

I really appreciate the quick reply, this worked out fine. Only mod I had to make was leveraging symbolic link due to the no exe requirement on the directory. You can call this one finished.
